lib/Runtime/Library/TypedArray.cpp

```diff
@@ -715,10 +758,49 @@ namespace Js
         AssertMsg(!PropertyRecord::IsPropertyNameNumeric(propertyNameString->GetString(), propertyNameString->GetLength()),
             "Numeric property names should have been converted to uint or PropertyRecord*");
 
+        ScriptContext *requestContext = GetScriptContext();
+        if (CanonicalNumericIndexString(propertyNameString, requestContext))
+        {
+            return FALSE;
```

Comment:
For typed arrays, the integer-indexed cases should flow through
BOOL TypedArrayBase::SetProperty(PropertyId propertyId, Var value, PropertyOperationFlags flags, PropertyValueInfo* info)
Seems like a good place for assertion + comment.

---
In reply to: [106301781](https://github.com/Microsoft/ChakraCore/pull/2640#discussion_r106301781) [](ancestors = 106301781)